google/cloud/forseti/services/inventory/storage.py

```diff
@@ -731,6 +665,97 @@ def get_inventory_indexes_older_than_cutoff(  # pylint: disable=invalid-name
         session.expunge_all()
         return inventory_indexes
 
+    @classmethod
+    def iter(cls,
+             session,
+             inventory_index_id,
+             type_list=None,
+             fetch_category=Categories.resource,
+             with_parent=False):
+        """Iterate the objects in the storage.
+
+        Args:
+            session (object): Database session.
+            inventory_index_id (str): the id of the inventory to open.
+            type_list (list): List of types to iterate over, or [] for all.
+            fetch_category (Categories): The category of data to fetch.
+            with_parent (bool): Join parent with results, yield tuples.
+
+        Yields:
+            object: Single row object or child/parent if 'with_parent' is set.
+        """
+        filters = [Inventory.inventory_index_id == inventory_index_id,
```

Comment:
This looks like an expression to me, will it preserve the query for you to use later on? Looks like it will get evaluated to a list of boolean i.e. [True, True]

Reply:
The == operator is overloaded for sqlAlchemy columns, so this produces an object, not a boolean: https://docs.sqlalchemy.org/en/13/core/tutorial.html#operators.